src/custom/hooks/usePriceImpact/useQuoteAndSwap.ts

```diff
@@ -34,6 +35,14 @@ type GetQuoteParams = {
 
 type FeeQuoteParamsWithError = FeeQuoteParams & { error?: QuoteError }
 
+function useOnSwapParamChange<T>({ onChangeCb, predicate }: { predicate: T; onChangeCb: (...params: any[]) => any }) {
+  const { typedValue, INPUT, OUTPUT, independentField } = useSwapState()
+
+  useEffect(() => {
+    Boolean(predicate) && onChangeCb
```

Comment:
I kind of get what if does, but i find it weird. If it's a conditional execution of a callback, Also, I find it better to use an IF.
Predicate has type `T` but shouldn't be boolean parameter? Also, more than a predicate seams like a flag which control wether the callback is called when the params change. If that's the case, maybe something more descriptive (`invokeCallbackOnChange: boolean`)

nit: `Boolean(predicate)` we usually do it as !!predicate